src/features/pastelPaperWalletGenerator/pastelPaperWalletGenerator.tsx

```diff
@@ -0,0 +1,177 @@
+import {
+  Document,
+  Page,
+  Path,
+  PDFViewer,
+  StyleSheet,
+  Svg,
+  Text,
+  View,
+} from '@react-pdf/renderer'
+import QRCode from 'qrcode.react'
+import React from 'react'
+import ReactDOMServer from 'react-dom/server'
+import Modal from 'react-modal'
+
+import styles from './PastelPaperWalletGenerator.module.css'
+
+type PastelPaperWalletGeneratorProps = {
+  address: string
+  privateKey: string
+  modalIsOpen: boolean
+  onCloseModal: () => void
+}
+
+function PastelPaperWalletGenerator(
+  props: PastelPaperWalletGeneratorProps,
+): JSX.Element {
+  console.log(props)
+  if (!props.modalIsOpen || !props.privateKey) {
+    return <></>
+  }
+
+  const breakChar = '\u00ad'
+  const pdfStyles = StyleSheet.create({
+    page: {
+      flexDirection: 'row',
+    },
+    section: {
+      margin: 0,
+      padding: 20,
+      flexGrow: 1,
+    },
+    contentTop: {
+      display: 'flex',
+      width: '100%',
+      marginBottom: '40px',
+      paddingBottom: '20px',
+      borderBottom: '2px dashed #000',
+      flexDirection: 'row',
+    },
+    contentItem: {
+      margin: '0',
+    },
+    contentTitle: {
+      marginTop: '0',
+      marginBottom: '5px',
+      fontSize: '18px',
+      fontWeight: 'bold',
+    },
+    contentValue: {
+      marginTop: '0',
+      fontSize: '12px',
+      fontWeight: 'normal',
+      wordBreak: 'break-all',
+    },
+    marginRight30: {
+      marginRight: '30px',
+    },
+    marginTop20: {
+      marginTop: '20px',
+    },
+    mainContent: {
+      display: 'flex',
+      flexDirection: 'row',
+      width: '100%',
+    },
+    topMedia: {
+      width: '35%',
+      maxWidth: '180px',
+      marginRight: '30px',
+    },
+    contentWrapper: {
+      width: '75%',
+    },
+    mainContentWrapper: {
+      width: '65%',
+    },
+    mainMedia: {
+      width: '35%',
+      maxWidth: '240px',
+      textAlign: 'right',
+    },
+    topImg: {
+      width: '100%',
+    },
+    mainImg: {
+      width: '100%',
+    },
+  })
+
+  function generateQrcode() {
+    const qrcodeValue = ReactDOMServer.renderToString(
+      <QRCode value={props.address} renderAs='svg' />,
+    )
+    const paths = qrcodeValue.match(/d="([^"]*)/g)
+    const fills = qrcodeValue.match(/fill="([^"]*)/g)
+
+    return (
+      <Svg
+        shape-rendering='crispEdges'
+        height='100%'
+        width='100%'
+        viewBox='0 0 33 33'
+      >
+        {paths &&
+          paths.length > 0 &&
+          paths?.map((path, idx) => (
+            <Path
+              fill={fills ? fills[idx].replace('fill="', '') : '#000'}
+              d={path.replace('d="', '')}
+              key={idx}
+            ></Path>
+          ))}
+      </Svg>
+    )
+  }
+
+  return (
+    <Modal
```

Comment:
When the modal opens, it expands for almost to the full width, and it's a bit difficult to close. Perhaps we can provide some closing mechanism?

Reply:
Added close button